src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,130 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.InvalidDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String DEFAULT_LOGIN = "user_login";
+    private static final String DEFAULT_PASSWORD = "user_password";
+    private static final long DEFAULT_ID = 123456789L;
+    private static final int DEFAULT_AGE = 23;
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_AGE = 18;
+    private static final int MAX_AGE = 122;
+
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+    private final StorageDaoImpl storageDao = new StorageDaoImpl();
+    private User user;
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setAge(DEFAULT_AGE);
+        user.setId(DEFAULT_ID);
+        user.setLogin(DEFAULT_LOGIN);
+        user.setPassword(DEFAULT_PASSWORD);
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_checkUser_ok() {
+        registrationService.register(user);
+        User actual = storageDao.get(user.getLogin());
+        assertEquals(actual, user);
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        user = null;
+        InvalidDataException invalidDataException = assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user));
+        assertEquals(invalidDataException.getMessage(),
+                "User cannot be NULL!");
+    }
+
+    @Test
+    void register_loginAlreadyExists_notOk() {
+        registrationService.register(user);
```

Comment:
put the first user directly into `Storage` bypassing the register method, because we should not use a method that is under a test to arrange the testing data + storage is a more low-level way